### ISSUE TITLE
Add basePath to allow multiple applications for the same proxy domain example.wip 

### ISSUE THREAD

**Issue by DracoBlue** [symfony-cli/symfony-cli#509]
(EDIT: maybe not necessary if #510 can handle this in a chained way - and it does not need a BC - see https://github.com/symfony-cli/symfony-cli/issues/509#issuecomment-2262350020 for reference)

Currently we can run:

```
symfony local:proxy:domain:attach example
```

or have a file like .symfony.local.yaml

```
proxy:
    domains:
        - example
```

and both make https://example.wip work with the current directory. The attach command currently supports --dir to specific which folder to use.

My suggestion is to add a `--basePath` which defaults to `/` and makes it possible to mount a different project in the same domain.

Here is an example:

* webapp folder `$ symfony local:proxy:domain:attach example`
* liveapp folder `$ symfony local:proxy:domain:attach example --basePath /live`

would mount all requests for **https://example.wip/*** to the **webapp** folder and all requests to **https://example.wip/live*** are executed in the **liveapp** folder.

The .symfony.local.yaml in **liveapp** could look like this:

```
proxy:
    domains:
        - domain: example
          basePath: /live
```

and the reader could fallback to using what is below the proxy.domains array as "domain" if it's just a string and not an object :).

Similiar (but different) request at: #335

**Comment by tucksaun** [comment]
> The attach command currently supports --dir to specific which folder to use.

Not exactly: `--dir` is a global flag used to define the project to work with. For example you can run `symfony server:stop --dir=/my/project` to stop the project while not having any shell opened in the project directory.

Regarding the `basePath` suggestion. This looks appealing from a UX standpoint ( I'm not a huge fan of the `basePath`, but this is a detail) and it might work.

But this would require heavy refactoring and BC code because the unicity of the a project per domain is the foundation of all this code.
This also adds questions: for example should be rewrite paths or not? What about ordering rules and overlapping rules?

This might be a lot of work for a potentially very niche feature.

**Comment by DracoBlue** [comment]
Thanks for this feedback @tucksaun !

It's good to know that the `--dir` is meant like this. But if for this example the attach command is executed in the webapp's folder or the liveapps folder the --dir option is not necessary and does not violate the usage.

A colleague of mine is really into GO-programming language and wants to help with this. I created also #510 for a similiar but very different use case. I guess the #510 has less breaking changes :). But I would let him try this - to scratch our own itch and keep you posted!

**Comment by DracoBlue** [comment]
The nice thing about #510 would be, we could emulate this behaviour by serving liveapp like this:

```
liveappfolder: $ symfony local:proxy:domain:attach example-live
```

and webbapp like this:

```
webappfolder: $ symfony local:proxy:domain:attach example
```

and then add a target like this:

```
webappfolder: $ symfony local:proxy:target:attach example --basePath /live --backend https://example-live.wip/live
```

This would mean #510 makes #509 also possible :).


I am just brainstorming :D
